homeassistant/components/huawei_lte/select.py

```diff
@@ -0,0 +1,119 @@
+"""Support for Huawei LTE selects."""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from functools import partial
+import logging
+from typing import Callable
+
+from huawei_lte_api.enums.net import LTEBandEnum, NetworkBandEnum
+
+from homeassistant.components.select import (
+    DOMAIN as SELECT_DOMAIN,
+    SelectEntity,
+    SelectEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity, EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HuaweiLteBaseEntityWithDevice
+from .const import DOMAIN, KEY_NET_NET_MODE
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up from config entry."""
+    router = hass.data[DOMAIN].routers[config_entry.unique_id]
+    switches: list[Entity] = []
+
+    desc = SelectEntityDescription(
+        key="network_mode", name="Preferred mode", entity_category=EntityCategory.CONFIG
+    )
+    switches.append(
+        HuaweiLteBaseSelect(
+            router,
+            description=desc,
+            key=KEY_NET_NET_MODE,
+            item="NetworkMode",
+            choices={
+                "00": "4G/3G/2G",
+                "01": "2G",
+                "02": "3G",
+                "03": "4G",
+                "0301": "4G/2G",
+                "0302": "4G/3G",
+                "0201": "3G/2G",
+            },
+            setter_method=partial(
+                router.client.net.set_net_mode,
+                LTEBandEnum.ALL,
+                NetworkBandEnum.ALL,
```

Comment:
These could be problematic in the sense that they'd reset what they were even in cases where that might not be required. I'm not sure if such a scenario actually exists, just a hunch.

Reply:
I would hope the backend library (or the device) handles it accordingly. Those are just wildcards for the corresponding network type, but I think I tested that changing between modes including 2G/3G/4G works.